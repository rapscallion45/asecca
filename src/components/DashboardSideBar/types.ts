/*
 * Dashboard Sidebar type definitions
 *
 * These type definitions are used in the applications Dashboard Sidebar component.
 */

/**
 * Nav Bar sidebar on close callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @callback IDashboardSidebarOnCloseCallback
 * @returns {void} - no return value
 */
export interface IDashboardSidebarOnCloseCallback {
  (): void;
}
