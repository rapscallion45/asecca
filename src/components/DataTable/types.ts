/*
 * Data Table type definitions
 *
 * These type definitions are used in the applications data table component.
 * Only make changes to this file if the corresponding change is made in the
 * Data Table component(s)
 */

/**
 * Data Table Row Cell Value types
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.13
 * @memberof DataTable
 *
 * @typedef DataTableRowCellValue
 */
export type DataTableRowCellValue =
  | string
  | Array<string>
  | boolean
  | null
  | undefined;

/**
 * Data Table Column types
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 * @memberof DataTable
 *
 * @typedef DataTableColumnType
 */
export type DataTableColumnType = 'string' | 'currency' | 'select' | 'check';

/**
 * Data Table Column properties
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef IDataTableColumn
 * @prop {string} label - data table column name
 * @prop {string} key - data table column data identifier
 * @prop {DataTableColumnType} type - data table column type
 */
export interface IDataTableColumn {
  label: string;
  key: string;
  type: DataTableColumnType;
}

/**
 * Data Table Row properties
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef IDataTableRow
 * @prop {string} label - data table row name
 */
export interface IDataTableRow {
  label: string;
}

/**
 * Data Table Cell Edit callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @callback IDataTableEditCellValueCallback
 * @param {DataTableRowCellValue} value - updated cell value, can be null
 * @param {string} colKey - cell column data identifier
 * @param {number} rowIdx - cell row index in data table
 * @returns {void} - no return value
 */
export interface IDataTableEditCellValueCallback {
  (value: DataTableRowCellValue, colKey: string, rowIdx: number): void;
}

/**
 * Data Table Cell Value Retrieval callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @callback IDataTableGetCellValueCallback
 * @param {number} rowIdx - cell row index in data table
 * @param {IDataTableColumn} column - cell column in data table
 * @returns {DataTableRowCellValue} - cell value, can be null or undefined
 */
export interface IDataTableGetCellValueCallback {
  (rowIdx: number, column: IDataTableColumn): DataTableRowCellValue;
}

/**
 * Data Table Currency Cell Value Edit callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @callback IDataTableEditCurrencyCellValueCallback
 * @param {string | null} value - updated cell value
 * @returns {void} - no return value
 */
export interface IDataTableEditCurrencyCellValueCallback {
  (value: string | null): void;
}

/**
 * Data Table Checkbox Cell Value Edit callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.13
 *
 * @callback IDataTableEditCheckboxCellValueCallback
 * @param {boolean | null} value - updated cell boolean value
 * @returns {void} - no return value
 */
export interface IDataTableEditCheckboxCellValueCallback {
  (value: boolean | null): void;
}
