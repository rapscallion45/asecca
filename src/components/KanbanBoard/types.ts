/*
 * Kanban Board type definitions
 *
 * These type definitions are used in the applications kanban board component.
 * Only make changes to this file if the corresponding change is made in the
 * Kanban Board component(s)
 */

/**
 * Kanban Board Column On Drop callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.7
 *
 * @callback IKanbanBoardColumnOnDropCallback
 * @param {number} colIndex - column index of task
 * @param {number} prevColIndex - previous column index of task
 * @param {number} taskIndex - index of task
 * @returns {void} - no return value
 */
export interface IKanbanBoardColumnOnDropCallback {
  (colIndex: number, prevColIndex: number, taskIndex: number): void;
}
