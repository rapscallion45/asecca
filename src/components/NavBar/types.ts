/*
 * Nav Bar type definitions
 *
 * These type definitions are used in the applications Nav Bar component.
 */

/**
 * Nav Bar sidebar open callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @callback INavBarOnSidebarOpenCallback
 * @returns {void} - no return value
 */
export interface INavBarOnSidebarOpenCallback {
  (): void;
}
