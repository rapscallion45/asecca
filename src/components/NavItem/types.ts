/*
 * Nav Item type definitions
 *
 * These type definitions are used in the applications Nav Item component.
 */

/**
 * Nav Item configuration properties
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 * @memberof NavItem
 *
 * @typedef INavItemConfig
 * @prop {string} title - nav item title text
 * @prop {string} path - nav item URL path
 * @prop {any} icon - nav item icon
 * @prop {boolean} newTab - nav item link opens in new tab
 * @prop {Array<INavItemConfig>} children - nav item child components
 */
export interface INavItemConfig {
  title: string;
  path?: string;
  icon?: any;
  newTab?: boolean;
  children?: Array<INavItemConfig>;
}
