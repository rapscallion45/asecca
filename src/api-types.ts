/*
 ** Costs Config return data type definition
 */
export type CostsConfigData = {
  application:
    | 'Per Device'
    | 'Per Batch'
    | 'Per Collection'
    | 'Per Project'
    | 'Per Destruction';
  collection_charge?: number | null;
  cost_source: 'Global' | 'Customer' | 'Project' | 'Collection';
  customer_charge?: number | null;
  effective_charge: number | null;
  global_charge?: number | null;
  line_type: 'Typical' | 'Custom';
  name: string;
  project_charge?: number | null;
};

/*
 ** Costs Config return data payload type definition
 */
export type CostsConfigDataPayload = {
  costs: Array<CostsConfigData>;
};

/*
 ** Proxy Error return data payload type definition
 */
export type ProxyErrorPayload = {
  message: string;
};
