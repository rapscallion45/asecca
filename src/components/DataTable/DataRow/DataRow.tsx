import { FC, useState, useEffect, ChangeEvent, useCallback } from 'react';
import { styled } from '@mui/material/styles';
import TableRow from '@mui/material/TableRow';
import { DataTableColumn } from '../types';
import CurrencyCell from './CurrencyCell/CurrencyCell';
import Cell from './Cell/Cell';

/* table row stylings */
const StyledTableRow = styled(TableRow)(({ theme }) => ({
  /* alternate row background colors */
  '&:nth-of-type(odd)': {
    backgroundColor: theme.palette.action.hover,
  },
  /* hide last border */
  '&:last-child td, &:last-child th': {
    border: 0,
  },
}));

interface DataTableRowProps {
  columns: Array<DataTableColumn>;
  row: any;
  editCol: any;
}

/* Data Table Row component */
/* ======================== */
const DataTableRow: FC<DataTableRowProps> = (props) => {
  const { row, columns, editCol } = props;

  /* initialise the edit value state to original column value */
  const [editValue, setEditValue] = useState<string>(
    row[editCol?.key] !== null ? parseFloat(row[editCol?.key]).toFixed(2) : '--'
  );

  /* flag in state for whether the edit column has been changed by user */
  const [isEdited, setIsEdited] = useState<boolean>(false);

  /* whenever the edit val changes, check if it is different from the original */
  useEffect(() => {
    if (
      row[editCol?.key] !== null &&
      editValue !== parseFloat(row[editCol?.key]).toFixed(2)
    ) {
      setIsEdited(true);
    } else if (
      row[editCol?.key] === null &&
      editValue !== '--' &&
      editValue !== ''
    ) {
      setIsEdited(true);
    } else {
      setIsEdited(false);
    }
  }, [editValue, row, editCol?.key]);

  /* callback for handling user input to the edit cell */
  const handleCurrencyValueChange = useCallback(
    (event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
      setEditValue(event.target.value);
    },
    []
  );

  const handleEditCellReformat = useCallback(() => {
    /* check if cell is null or indicating null */
    if (editValue === '' || editValue === '--') {
      /* leave cell as null input indication */
      setEditValue('--');
      return;
    }

    /* check if user input is number, if so, format correctly */
    if (/^(\d+.)*(\d+)$/.test(editValue)) {
      setEditValue(parseFloat(editValue).toFixed(2));
    } else {
      /* user entered non-number, ignore input */
      setEditValue(
        row[editCol?.key] !== null
          ? parseFloat(row[editCol?.key]).toFixed(2)
          : '--'
      );
    }
  }, [row, editCol?.key, editValue]);

  const handleEditCellOnClick = useCallback(() => {
    /* check if cell is currently null or indicating null */
    if (editValue === '--' && !isEdited)
      /* clear cell if null, ready for new input */
      setEditValue('');
  }, [editValue, isEdited]);

  const handleResetCell = useCallback(() => {
    /* user has decided to clear their edits, restore cell to orig value */
    setEditValue(
      row[editCol?.key] !== null
        ? parseFloat(row[editCol?.key]).toFixed(2)
        : '--'
    );
  }, [row, editCol?.key]);

  return (
    <StyledTableRow>
      {/* map passed column data for current row */}
      {columns.map((column: any) =>
        column.key !== 'name' ? (
          <CurrencyCell
            key={`${row.name}-${column.key}`}
            canEdit={column.label === editCol?.label}
            isEdited={isEdited}
            row={row}
            column={column}
            editValue={editValue}
            handleEditValueChange={handleCurrencyValueChange}
            handleEditValueReformat={handleEditCellReformat}
            handleEditCellOnClick={handleEditCellOnClick}
            handleResetCell={handleResetCell}
          />
        ) : (
          <Cell value={row[column.key]} />
        )
      )}
    </StyledTableRow>
  );
};

export default DataTableRow;
