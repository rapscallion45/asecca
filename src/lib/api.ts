import base64 from 'base-64';
import queryString from 'query-string';
import { ICostsConfigSaveDataPayload } from '@/api-types';

/* configuration params for staging DB */
const { STAGING_DB_USERNAME, STAGING_DB_PASSWORD, STAGING_DB_REST_API_URL } =
  process.env;

async function getCostsConfig(
  query: Partial<{
    [key: string]: string | string[];
  }>
) {
  /* setup GET request options with basic auth */
  const requestOptions = {
    method: 'GET',
    headers: new Headers({
      'Content-Type': 'application/json',
      Authorization: `Basic ${base64.encode(
        `${STAGING_DB_USERNAME}:${STAGING_DB_PASSWORD}`
      )}`,
    }),
  };

  /* construct query param string from passed query object */
  const queryParamString = queryString.stringify(query);

  /* fetch the costs config */
  return fetch(
    `${STAGING_DB_REST_API_URL}/api/costs_config?${queryParamString}`,
    requestOptions
  );
}

async function setCostsConfig(body: ICostsConfigSaveDataPayload) {
  /* setup POST request options with basic auth */
  const requestOptions = {
    method: 'POST',
    headers: new Headers({
      'Content-Type': 'application/json',
      Authorization: `Basic ${base64.encode(
        `${STAGING_DB_USERNAME}:${STAGING_DB_PASSWORD}`
      )}`,
    }),
    body: JSON.stringify(body),
  };

  /* send the costs config */
  return fetch(`${STAGING_DB_REST_API_URL}/api/costs_config`, requestOptions);
}

const api = {
  getCostsConfig,
  setCostsConfig,
};

export default api;
