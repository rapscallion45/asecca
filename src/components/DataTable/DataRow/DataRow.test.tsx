import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Table, TableBody } from '@mui/material';
import { ICostsConfigData } from '@/lib/api/api-types';
import DataRow from './DataRow';
import dataTableColumns from '../../CostsConfigTable/costsConfigTableColumns';
import costsConfigDataMock from '../../../../__mocks__/Configurators/costsConfigDataMock';
import { IDataTableColumn } from '../types';

/* mock callbacks for get and edit cell values */
const mockGetCellValueCallback = jest.fn(
  (rowIdx: number, column: IDataTableColumn) =>
    costsConfigDataMock.costs[rowIdx][column.key as keyof ICostsConfigData]
);
const mockEditCellValueCallback = jest.fn(
  (value: string | null, colKey: string, rowIdx: number) =>
    costsConfigDataMock.costs[rowIdx][colKey as keyof ICostsConfigData]
);

/**
 * Data Table Data Row Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 */
describe('Data Table Data Row', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={0}
              columns={dataTableColumns}
              editableColLabels={['Collection']}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Value', () => {
    /** ensure mock function calls are cleared after test */
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('Should render expected value for row and column', async () => {
      /** Arrange */
      const testRowIdx: number = 0;
      const testEditCol: string = 'Global';
      const testEditColKey: string = 'global_charge';

      /** Act */
      render(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={testRowIdx}
              columns={dataTableColumns}
              editableColLabels={[testEditCol]}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      );

      /** Assert - correct value should be displayed from test data */
      expect(
        screen.getByDisplayValue(
          costsConfigDataMock.costs[testRowIdx][
            testEditColKey as keyof ICostsConfigData
          ] as string
        )
      ).toBeInTheDocument();
    });

    it('Should render null indicator if null for this row and column', async () => {
      /** Arrange */
      const testRowIdx: number = 0;
      const testEditCol: string = 'Collection';

      /** Act */
      render(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={testRowIdx}
              columns={dataTableColumns}
              editableColLabels={[testEditCol]}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      );

      /** Assert - passed value is null, therefore clear icon should not render */
      expect(screen.getByDisplayValue('--')).toBeInTheDocument();
    });

    it('Should render Currency Cell for column type curreny', async () => {
      /** Arrange */
      const testRowIdx: number = 0;
      const testEditCol: string = 'Collection';
      const testColumn: Array<IDataTableColumn> = dataTableColumns.filter(
        (column) => column.label === 'Collection'
      );

      /** Act */
      render(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={testRowIdx}
              columns={testColumn}
              editableColLabels={[testEditCol]}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      );

      /** Assert - currency cell will display currency symbol */
      expect(screen.getByText('£')).toBeInTheDocument();
    });

    it('Should render normal Cell for non curreny type', async () => {
      /** Arrange */
      const testRowIdx: number = 0;
      const testEditCol: string = 'Product';
      const testColumn: Array<IDataTableColumn> = dataTableColumns.filter(
        (column) => column.label === 'Product'
      );

      /** Act */
      render(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={testRowIdx}
              columns={testColumn}
              editableColLabels={[testEditCol]}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      );

      /** Assert - currency cell should not be rendered */
      expect(screen.queryByText('£')).toBeNull();
    });

    it('Should submit passed value with row index and column key', async () => {
      /** Arrange */
      const testInput: string = '128';
      const testFormattedInput: string = '128.00';
      const testRowIdx: number = 0;
      const testEditCol: string = 'Global';
      const testEditColKey: string = 'global_charge';
      const assertValue: string = costsConfigDataMock.costs[testRowIdx][
        testEditColKey as keyof ICostsConfigData
      ] as string;

      /** Act */
      render(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={testRowIdx}
              columns={dataTableColumns}
              editableColLabels={[testEditCol]}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      );

      /** Assert - input field, button, and clear icon should be rendered */
      expect(screen.getByDisplayValue(assertValue)).toBeInTheDocument();
      expect(screen.queryByRole('button')).toBeInTheDocument();
      expect(screen.queryByTestId('CloseIcon')).toBeInTheDocument();

      /** Act - click on input field, update value and press enter */
      fireEvent(
        screen.getByDisplayValue(
          costsConfigDataMock.costs[testRowIdx][
            testEditColKey as keyof ICostsConfigData
          ] as string
        ),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
      fireEvent.change(
        screen.getByDisplayValue(
          costsConfigDataMock.costs[testRowIdx][
            testEditColKey as keyof ICostsConfigData
          ] as string
        ),
        {
          target: { value: testInput },
        }
      );

      /** Assert - wait for input to update */
      await waitFor(() => {
        expect(screen.getByDisplayValue(testInput)).toBeInTheDocument();
      });

      /** Act - press enter key */
      fireEvent.keyDown(screen.getByDisplayValue(testInput), {
        key: 'Enter',
        code: 'Enter',
        keyCode: 13,
        charCode: 13,
      });

      /** Assert - button and clear icon not rendered, values updated */
      await waitFor(() => {
        /** display value of cell updated to formatted value */
        expect(
          screen.getByDisplayValue(testFormattedInput)
        ).toBeInTheDocument();
      });
      /** edit cell callback to have been called once, with formatted value */
      expect(mockEditCellValueCallback.mock.calls).toHaveLength(1);
      expect(mockEditCellValueCallback.mock.calls[0][0]).toBe(
        testFormattedInput
      );
      /** edit cell callback to have been called with correct column key */
      expect(mockEditCellValueCallback.mock.calls[0][1]).toBe(testEditColKey);
      /** edit cell callback to have been called with correct column key */
      expect(mockEditCellValueCallback.mock.calls[0][2]).toBe(testRowIdx);
    });

    it('Should submit null with row and column if null indicated for cell', async () => {
      /** Arrange */
      const testInput: string = '';
      const testFormattedInput: string = '--';
      const testRowIdx: number = 0;
      const testEditCol: string = 'Global';
      const testEditColKey: string = 'global_charge';
      const assertValue: string = costsConfigDataMock.costs[testRowIdx][
        testEditColKey as keyof ICostsConfigData
      ] as string;

      /** Act */
      render(
        <Table>
          <TableBody>
            <DataRow
              rowName="testrow"
              rowIdx={testRowIdx}
              columns={dataTableColumns}
              editableColLabels={[testEditCol]}
              getCellValueCallback={mockGetCellValueCallback}
              editCellValueCallback={mockEditCellValueCallback}
            />
          </TableBody>
        </Table>
      );

      /** Assert - input field, button, and clear icon should be rendered */
      expect(screen.getByDisplayValue(assertValue)).toBeInTheDocument();
      expect(screen.queryByRole('button')).toBeInTheDocument();
      expect(screen.queryByTestId('CloseIcon')).toBeInTheDocument();

      /** Act - click on input field, update value and press enter */
      fireEvent(
        screen.getByDisplayValue(
          costsConfigDataMock.costs[testRowIdx][
            testEditColKey as keyof ICostsConfigData
          ] as string
        ),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
      fireEvent.change(
        screen.getByDisplayValue(
          costsConfigDataMock.costs[testRowIdx][
            testEditColKey as keyof ICostsConfigData
          ] as string
        ),
        {
          target: { value: testInput },
        }
      );

      /** Assert - wait for input to update */
      await waitFor(() => {
        expect(screen.getByDisplayValue(testInput)).toBeInTheDocument();
      });

      /** Act - press enter key */
      fireEvent.keyDown(screen.getByDisplayValue(testInput), {
        key: 'Enter',
        code: 'Enter',
        keyCode: 13,
        charCode: 13,
      });

      /** Assert - button and clear icon not rendered, values updated */
      await waitFor(() => {
        /** display value of cell updated to formatted value */
        expect(
          screen.getByDisplayValue(testFormattedInput)
        ).toBeInTheDocument();
      });
      /** edit cell callback to have been called once, with null value */
      expect(mockEditCellValueCallback.mock.calls).toHaveLength(1);
      expect(mockEditCellValueCallback.mock.calls[0][0]).toBe(null);
      /** edit cell callback to have been called with correct column key */
      expect(mockEditCellValueCallback.mock.calls[0][1]).toBe(testEditColKey);
      /** edit cell callback to have been called with correct column key */
      expect(mockEditCellValueCallback.mock.calls[0][2]).toBe(testRowIdx);
    });
  });
});
