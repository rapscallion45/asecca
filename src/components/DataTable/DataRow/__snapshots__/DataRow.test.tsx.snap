// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Data Row Renders correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1enotpu-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-j0r6z9-MuiTableCell-root"
      >
        Device Processing
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-ujx7iw-MuiTableCell-root"
      >
        Per Device
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-nx6mre-MuiTableCell-root"
      >
        £9.00
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-nx6mre-MuiTableCell-root"
      >
        --
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-nx6mre-MuiTableCell-root"
      >
        --
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1mlukn3-MuiTableCell-root"
      >
        <div
          className="MuiFormControl-root css-e363es-MuiFormControl-root"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-vg7q19-MuiInputBase-root-MuiInput-root"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            required={true}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
              >
                £
              </p>
            </div>
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
              disabled={false}
              id="testrow-collection_charge-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={true}
              type="text"
              value="--"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="clear user entry"
                className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              />
            </div>
          </div>
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1nt70hm-MuiTableCell-root"
      >
        £Global
      </td>
    </tr>
  </tbody>
</table>
`;
